product-mini/platforms/linux-sgx/enclave-sample/App/App.cpp

```diff
@@ -734,3 +739,136 @@ main(int argc, char *argv[])
+        return 1;
+    }
+} 
+
+int
+wamr_pal_create_process(struct wamr_pal_create_process_args *args) {
+    uint32_t stack_size = 16 * 1024, heap_size = 16 * 1024;
+    int log_verbose_level = 2;
+    bool is_repl_mode = false, alloc_with_pool = false;
+    const char *dir_list[8] = { NULL };
+    uint32_t dir_list_size = 0;
+    const char *env_list[8] = { NULL };
+    uint32_t env_list_size = 0;
+    uint32_t max_thread_num = 4;
+    
+    int argc = 2;
+    char *argv[argc] = { (char*)"./iwasm", (char *)args->argv[0]};
+    
+    uint8_t *wasm_files_buf = NULL;
+    void *wasm_modules = NULL;
+    int len=0;
+    
+    char *temp = (char *)args->argv[0];
+    while(temp){
+        len++;
+        temp=(char *)args->argv[len];
+    }
+    char *wasm_files[len];
+    void *wasm_module_inst[len];
+    for (int i=0; i<len; ++i) {
+        wasm_files[i] = (char *)args->argv[i];
+    }
+
+    /* Init runtime */
+    if (!init_runtime(alloc_with_pool, max_thread_num)) {
+	printf("Failed to init runtime\n");
+        return -1;
+    }
+
+    /* Set log verbose level */
+    if (!set_log_verbose_level(log_verbose_level)) {
+        printf("Failed to set log level\n");
+        return -1;
+    }
+
+    for (int i=0; i<len; ++i) {
+        uint8_t *wasm_file_buf = NULL;
+        uint32_t wasm_file_size;
+        void *wasm_module = NULL;
+        char error_buf[128] = { 0 };
+        /* Load WASM byte buffer from WASM bin file */
+        if (!(wasm_file_buf =
+            (uint8_t *)read_file_to_buffer(wasm_files[i], &wasm_file_size))) {
+                printf("Failed to read file to buffer\n");
+                return -1;
+        }
+
+        /* Load module */
+        if (!(wasm_module = load_module(wasm_file_buf, wasm_file_size,
+                                    error_buf, sizeof(error_buf)))) {
+                printf("%s\n", error_buf);
+		printf("Failed to load buffer to wams module\n");
+                return -1;
+        }
+
+        /* Set wasi arguments */
+        if (!set_wasi_args(wasm_module, dir_list, dir_list_size,
+                               env_list, env_list_size, argv, argc)) {
+                printf("%s\n", "set wasi arguments failed.\n");
+		printf("Failed to set wasi arguments\n");
+                return -1;
+        }
+
+        /* Instantiate module */
+        if (!(wasm_module_inst[i] = instantiate_module(wasm_module,
+                                                stack_size, heap_size,
+                                                error_buf,
+                                                sizeof(error_buf)))) {
+                printf("%s\n", error_buf);
+		printf("Failed to read file to buffer\n");
+                return -1;
+        }
+
+
+        app_instance_main(wasm_module_inst[i], argc, argv);
+
+        /* Deinstantiate module */
+        deinstantiate_module(wasm_module_inst[i]);
+    }
+    return 0;
+}
+
+
+int wamr_pal_destroy(void) {
+    destroy_runtime();
```

Comment:
if wamr_pal_create_process destroys runtime, here we don't need to destroy again.

Reply:
We need destroy the runtime in wamr_pal_destroy, so in pal_create_process we only free the file buffer and unload the module.